src/main/resources/com/google/api/codegen/csharp/gapic_snippets.snip

```diff
@@ -0,0 +1,229 @@
+@extends "csharp/common.snip"
+
+@snippet generate(api)
+    // Copyright 2016 Google Inc. All Rights Reserved.
+    //
+    // Licensed under the Apache License, Version 2.0 (the "License");
+    // you may not use this file except in compliance with the License.
+    // You may obtain a copy of the License at
+    //
+    //     http://www.apache.org/licenses/LICENSE-2.0
+    //
+    // Unless required by applicable law or agreed to in writing, software
+    // distributed under the License is distributed on an "AS IS" BASIS,
+    // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    // See the License for the specific language governing permissions and
+    // limitations under the License.
+
+    // Generated code. DO NOT EDIT!
+
+    @join import : api.imports
+        using {@import.fullName};
+    @end
+
+    namespace {@api.packageName}
+    {
+        {@snippets(api)}
+    }
+
+@end
+
+@private snippets(api)
+    public class {@api.name}
+    {
+        @join method : api.snippetMethods
+            {@method(api, method)}
+
+        @end
+    }
+@end
+
+@private method(api, method)
+    @switch method.type.toString
+    @case "FlattenedMethod"
+        {@flattenedMethodSync(api, method)}
+    @case "FlattenedAsyncCallSettingsMethod"
+        {@flattenedMethodAsyncCallSettings(api, method)}
+    @case "PagedFlattenedMethod"
+        {@pagedFlattenedMethodSync(api, method)}
+    @case "PagedFlattenedAsyncMethod" 
+        {@pagedFlattenedMethodAsync(api, method)}
+    @default
+        *** ERROR: Cannot handle method type '{@method.type.toString}' ***
```

Comment:
I'd prefer to have the unhandled cases consistent, so we can do an automated check on them appearing in the output. The existing pattern is the one you used in renderInitValue below:

`$unhandledCase: {@initValue.type}$`